go/vt/sqlparser/sql.y

```diff
@@ -5682,7 +5699,7 @@ match_option:
  }
 
 charset:
-  id_or_var
+  sql_id
```

Comment:
This is more consistent with how we parse this in other places, directly allowing non reserved keywords like `ASCII` in this case.